### PR TITLE
Web.JSON: Encode JSON with indents

### PR DESCRIPTION
**Pull request by lambdalisue**
With this changes, `s:JSON.encode({obj}, { 'indent': 2 })` encode `{obj}` into a JSON string **with indents**.

```vim
echo s:JSON.encode({'a': 0, 'b': 1})
" => '{"a":0,"b":1}'
echo s:JSON.encode({'a': 0, 'b': 1}, {'indent': 2})
" => '{
"   "a": 0,
"   "b": 1
" }'
```

### CODE REVIEW

File: doc/vital-web-json.txt [hunk 1]
```diff
@@ -41,31 +43,44 @@ null					*Vital.Web.JSON.null*
...
 	(e.g. |Vital.Web.JSON.true|) are encoded into corresponding javascript
 	tokens (e.g. 'true').
 >
-    " it will echo '[true, false, null]'
-    echo s:JSON.encode([s:JSON.true, s:JSON.false, s:JSON.null])
->
+	echo s:JSON.encode([s:JSON.true, s:JSON.false, s:JSON.null])
+	" => '[true, false, null]'
+<
+	{settings} is a |Dictionary| which allow the following:
```
Comment (ujihisa) [inline doc/vital-web-json.txt]: allows

File: doc/vital-web-json.txt [hunk 2]
```diff
@@ -41,31 +43,44 @@ null					*Vital.Web.JSON.null*
...
 
-decode({json} [, {settings}])  		*Vital.Web.JSON.decode()*
+	'indent'
+	The number of spaces used as an indent. When the value is grater than
```
Comment (ujihisa) [inline doc/vital-web-json.txt]: an indentation.

Comment (ujihisa) [inline doc/vital-web-json.txt]: s/[grater](https://www.google.ca/search?q=grater&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiNsc7a--XJAhUT8mMKHQIyB8EQ_AUIBygB&biw=1322&bih=895)/greater/

File: doc/vital-web-json.txt [hunk 3]
```diff
@@ -41,31 +43,44 @@ null					*Vital.Web.JSON.null*
...
+<
+decode({json}[, {settings}])		*Vital.Web.JSON.decode()*
 	Decode a JSON string into an object that vim can treat.
-	{settings} is a |Dictionary| which contains the following items:
-
-	"use_token"	    Default: 0
-		Use special tokens (e.g. |Vital.Web.JSON.true|) to
-		represent coresponding javascript tokens (e.g. 'true').
-		Otherwise 1 or 0 are used to represent these.
+	{settings} is a |Dictionary| which allow the following:
```
Comment (ujihisa) [inline doc/vital-web-json.txt]: allows

File: doc/vital-web-json.txt [hunk 4]
```diff
@@ -41,31 +43,44 @@ null					*Vital.Web.JSON.null*
...
 
+	'use_token'
+	Use special tokens (e.g. |Vital.Web.JSON.true|) to represent coresponding
```
Comment (ujihisa) [inline doc/vital-web-json.txt]: corresponding

**Comment by lambdalisue** [comment]
Let me know if I need to write `Changes`.

**Comment by ujihisa** [comment]
Yes. The argument that function accept changed.

**Comment by ujihisa** [comment]
:+1: :sushi: for me.

Please merge by yourself, after joining to the vital.vim Collaborator team.

**Comment by lambdalisue** [comment]
> Please merge by yourself, after joining to the vital.vim Collaborator team.

Am I going to? lol

**Comment by thinca** [comment]
This PR keeps compatibility.  So you don't need to change `Changes`.

**Comment by lambdalisue** [comment]
Ok. I removed the commit.

**Comment by thinca** [comment]
LGTM :+1:

**Comment by lambdalisue** [comment]
Thanks!

**Pull request merged**